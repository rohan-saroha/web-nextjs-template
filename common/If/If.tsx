import React, { PropsWithChildren } from "react";

interface Props {
  otherwise?: React.ReactElement;
  condition: any;
}

const If = (props: PropsWithChildren<Props>) => {
  if (props.condition) return <>{props.children}</>;
  return <>{props.otherwise || null}</>;
};

export default If;
