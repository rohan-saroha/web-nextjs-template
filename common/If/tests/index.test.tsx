import React from "react";
import { render } from "@testing-library/react";
import If from "../If";

describe("<If />", () => {
  it("should render and match the snapshot", () => {
    const { baseElement } = render(<If condition={false} otherwise={<div>Should render</div>} />);
    expect(baseElement).toMatchSnapshot();
  });
});
