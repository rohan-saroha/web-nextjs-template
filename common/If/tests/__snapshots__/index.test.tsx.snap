// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<If /> should render and match the snapshot 1`] = `
<body>
  <div>
    <div>
      Should render
    </div>
  </div>
</body>
`;
